### PR TITLE
EOS-22963: Check message delivery status

### PR DESCRIPTION
**Pull request by selvakumaar5496**
Signed-off-by: Selvakumar <selva.k.nambi@seagate.com>

# Problem Statement
- https://jts.seagate.com/browse/EOS-22963

# Design
-  Added a callback function to check status of message delivery

# Coding 
-  Coding conventions are followed and code is consistent [Y/N]: Y
-  Confirm All CODACY errors are resolved [Y/N]: Y

# Testing 
- [x] Confirm that Test Cases are added (for both the cases, fix and feature) [Y/N]: #410
- [x] Confirm Test Cases cover Happy Path, Non-Happy Path and Scalability [Y/N]: #410
- [x] Confirm Testing was performed with installed RPM [Y/N]:  Y (Also tested with the machine where bug was reported)

# Review Checklist 
  Before posting the PR please ensure
- [x] PR is self reviewed
- [ ] Is there a change in filename/package/module or signature [Y/N]: N
- [ ] If yes for above point, Is a notification sent to all other cortx components [Y/N] N
- [x] Jira is updated
- [x] Check if the description is clear and explained. 
- [x] Check Acceptance Criterion is defined. 
- [ ] All the tests performed should be mentioned before Resolving a JIRA. 
- [ ] Verification needs to be done before marked as Closed/Verified 

# Documentation
- [ ] Changes done to WIKI / Confluence page

### CODE REVIEW

File: py-utils/src/utils/message_bus/message_broker_collection.py [hunk 1]
```diff
@@ -50,6 +50,9 @@ class KafkaMessageBroker(MessageBroker):
     # Socket timeout
     _kafka_socket_timeout = 15000
 
+    # Message timeout
+    _message_timeout = 5000
```
Comment (ujjwalpl) [inline py-utils/src/utils/message_bus/message_broker_collection.py:54]: Why is it hard coded ?

File: py-utils/src/utils/message_bus/message_broker_collection.py [hunk 2]
```diff
@@ -322,7 +332,8 @@ def send(self, producer_id: str, message_type: str, method: str, \
                 "Producer %s is not initialized", producer_id)
 
         for message in messages:
-            producer.produce(message_type, bytes(message, 'utf-8'))
+            producer.produce(message_type, bytes(message, 'utf-8'), \
+                callback=self.delivery_callback)
```
Comment (ujjwalpl) [inline py-utils/src/utils/message_bus/message_broker_collection.py:336]: delivery_callback is static method but the call is with self. May have issues. 

Also, I dont see any test code.

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/src/utils/message_bus/message_broker_collection.py [hunk 3]
```diff
@@ -50,6 +50,9 @@ class KafkaMessageBroker(MessageBroker):
     # Socket timeout
     _kafka_socket_timeout = 15000
```
Comment (sachinpunadikar) [inline py-utils/src/utils/message_bus/message_broker_collection.py:51]: rename this to generic name like socket_timeout

Comment (selvakumaar5496) [reply py-utils/src/utils/message_bus/message_broker_collection.py:51]: Done

File: py-utils/src/utils/message_bus/message_broker_collection.py (hunk 2 above)
Comment (selvakumaar5496) [reply py-utils/src/utils/message_bus/message_broker_collection.py:336]: image

Should be called with self. But delivery_callback method doesnt take self as argument.

File: py-utils/src/utils/message_bus/message_broker_collection.py (hunk 1 above)
Comment (selvakumaar5496) [reply py-utils/src/utils/message_bus/message_broker_collection.py]: Reading it from config file.

File: py-utils/src/utils/message_bus/message_broker_collection.py (hunk 2 above)
Comment (selvakumaar5496) [reply py-utils/src/utils/message_bus/message_broker_collection.py:334]: Test code is included in #410

**Review by sachinpunadikar — APPROVED** [review]
+1

File: py-utils/src/setup/templates/utils.config.tmpl.1-node [hunk 4]
```diff
@@ -6,6 +6,10 @@ cortx:
       servers:
       # Value should be in format <FQDN>:<port>. Port is optional
       - TMPL_KAFKA_SERVER1_FQDN:TMPL_KAFKA_SERVER1_PORT
+      config:
```
Comment (ujjwalpl) [inline py-utils/src/setup/templates/utils.config.tmpl.1-node:9]: Timeouts can be generic timeouts and not specific to kafka and can be used and mapped to kafka. 
```
kafka:
    servers:
message_bus: 
    default_timeout: 2
    socket_timeout: 15000
    message_timeout: 5000
```
Also need to expand timeout properly like it is not clear what is default_timeout or message_timeout.

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/src/setup/templates/utils.config.tmpl.1-node [hunk 5]
```diff
@@ -6,6 +6,10 @@ cortx:
...
+        default_timeout: 2
```
Comment (ujjwalpl) [inline py-utils/src/setup/templates/utils.config.tmpl.1-node:10]: Where is default_timeout used in the code ?

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/src/setup/templates/utils.config.tmpl.1-node (hunk 5 above)
Comment (selvakumaar5496) [reply py-utils/src/setup/templates/utils.config.tmpl.1-node]: used in consumer.poll()

File: py-utils/src/setup/templates/utils.config.tmpl.1-node (hunk 4 above)
Comment (selvakumaar5496) [reply py-utils/src/setup/templates/utils.config.tmpl.1-node]: Added with comments

File: py-utils/src/setup/utils.py [hunk 6]
```diff
@@ -53,16 +53,18 @@ def _get_utils_path() -> str:
...
 
         with open(r'/etc/cortx/message_bus.conf.sample', 'w+') as file:
             json.dump({}, file, indent=2)
         Conf.load('index', 'json:///etc/cortx/message_bus.conf.sample')
         Conf.set('index', 'message_broker>type', 'kafka')
-        for i in range(len(kafka_server_list)):
+        for i in range(len(message_server_list)):
             Conf.set('index', f'message_broker>cluster[{i}]', \
-                {'server': kafka_server_list[i], 'port': port_list[i]})
+                {'server': message_server_list[i], 'port': port_list[i]})
```
Comment (ujjwalpl) [inline py-utils/src/setup/utils.py:66]: We can not / should not set value as dict. It should be key (string) and value (string)

Conf.set('index', f'message_broker>cluster[{i}]>server', message_server_list[i])
Conf.set('index', f'message_broker>cluster[{i}]>post', port_list[i])

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/src/setup/utils.py (hunk 6 above)
Comment (selvakumaar5496) [reply py-utils/src/setup/utils.py]: Done with above suggested way

**Review by ujjwalpl — APPROVED** [review]

**Pull request merged**